src/status_im/browser/core.cljs

```diff
@@ -403,18 +403,39 @@
                                                   :error     %1
                                                   :result    %2}])]}))))
 
+(fx/defn handle-no-permissions [cofx {:keys [method id]} message-id]
+  (if (= method "eth_accounts")
+    ;; eth_accounts returns empty array for compatibility with meta-mask
+    (send-to-bridge cofx
+                    {:type      constants/web3-send-async-callback
+                     :messageId message-id
+                     :result    {:jsonrpc "2.0"
+                                 :id      (int id)
+                                 :result  []}})
+    (send-to-bridge cofx
+                    {:type      constants/web3-send-async-callback
+                     :messageId message-id
+                     :error     {:code 4100}})))
+
+
+(def permissioned-methods
+  #{"eth_accounts" "eth_coinbase" "eth_sendTransaction" "eth_sign"
+    "keycard_signTypedData"
+    "eth_signTypedData" "personal_sign" "personal_ecRecover"})
+
+(def require-permission?
+  (comp boolean permissioned-methods))
+
+(defn has-permissions? [{:dapps/keys [permissions]} dapp-name method]
+  (boolean
+   (and (require-permission? method)
+        (not (some #{constants/dapp-permission-web3} (get-in permissions [dapp-name :permissions]))))))
```

Comment:
This can simply be:

```clojure
(and (permissioned-methods method)
     (not (some ...)))
```

You don't need to typecast manually. Is there any specific reason to add an extra `def require-permission?`